packages/slice-machine/src/features/builder/fields/ContentRelationshipFieldPicker.tsx

```diff
@@ -1167,26 +1173,60 @@ function createContentRelationshipLinkCustomtypes(
  * Generic recursive function that goes down the fields check map and counts all
  * the properties that are set to true, which correspond to selected fields.
  *
+ * Distinguishes between all picked fields and nested picked fields within a
+ * content relationship field.
+ *
  * It's not type safe, but checks the type of the values at runtime so that
  * it only recurses into valid objects, and only counts checkbox fields.
  */
-function countPickedFields(
+type CountPickedFieldsResult = {
+  pickedFields: number;
+  nestedPickedFields: number;
+};
+
+export function countPickedFields(
   fields: Record<string, unknown> | undefined,
-): number {
-  if (!fields) return 0;
-  return Object.values(fields).reduce<number>((count, value) => {
-    if (!isValidObject(value)) return count;
-    if (isCheckboxValue(value)) return count + (value.value ? 1 : 0);
-    return count + countPickedFields(value);
-  }, 0);
-}
-function isCheckboxValue(value: unknown): value is PickerCheckboxField {
-  if (!isValidObject(value)) return false;
-  return "type" in value && value.type === "checkbox";
-}
+  isNested = false,
+): CountPickedFieldsResult {
+  if (!fields) return { pickedFields: 0, nestedPickedFields: 0 };
+
+  return Object.values(fields).reduce<CountPickedFieldsResult>(
+    (result, value) => {
+      if (!isValidObject(value)) return result;
+
+      if ("type" in value && value.type === "checkbox") {
+        const isChecked = Boolean(value.value);
+        if (!isChecked) return result;
+
+        return {
+          pickedFields: result.pickedFields + 1,
+          nestedPickedFields: result.nestedPickedFields + (isNested ? 1 : 0),
+        };
+      }
+
+      if ("type" in value && value.type === "contentRelationship") {
+        const { pickedFields } = countPickedFields(value, isNested);
+        const { nestedPickedFields } = countPickedFields(value, true);
+
+        return {
+          pickedFields: result.pickedFields + pickedFields,
+          nestedPickedFields: result.nestedPickedFields + nestedPickedFields,
+        };
+      }
 
-function isGroupField(field: NestableWidget | Group): field is Group {
-  return field.type === "Group";
+      const { pickedFields } = countPickedFields(value, isNested);
+      const { nestedPickedFields } = countPickedFields(value, isNested);
```

Comment:
❓ Same as https://github.com/prismicio/slice-machine/pull/1672/files#r2164620666, would this work the same?

```suggestion
      const { pickedFields, nestedPickedFields } = countPickedFields(value, isNested);
```

Reply:
😅 I rework that and forgot about it, thanks!